src/main/java/com/yahoo/bullet/kafka/CertRefreshingSslEngineFactory.java

```diff
@@ -0,0 +1,205 @@
+/*
+ *  Copyright 2021, Yahoo Inc.
+ *  Licensed under the terms of the Apache License, Version 2.0.
+ *  See the LICENSE file associated with the project for terms.
+ */
+package com.yahoo.bullet.kafka;
+
+import com.oath.auth.KeyRefresher;
+import com.oath.auth.Utils;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.common.config.SslConfigs;
+import org.apache.kafka.common.config.types.Password;
+import org.apache.kafka.common.security.auth.SslEngineFactory;
+import org.apache.kafka.common.utils.SecurityUtils;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLParameters;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.security.KeyStore;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+@Slf4j
+public class CertRefreshingSslEngineFactory implements SslEngineFactory {
+    public static final String SSL_CERT_LOCATION_CONFIG = "ssl.cert.refreshing.cert.location";
```

Comment:
Let's move these to KafkaConfig and add the required validations for them if present. Add default values to KafkaConfig with `: ` as values (null)